wp/lib/bap_wp/src/precondition.ml

```diff
@@ -401,9 +401,16 @@ let input_regs (arch : Arch.t) : Var.t list =
        causes Z3 to slow down during evaluation. *)
     info "[mem] is not included as an input to the function call.%!";
     [rdi; rsi; rdx; rcx; r.(0); r.(1)]
-  | _ ->
-    raise (Not_implemented "input_regs: Input registers have not been \
-                            implemented for non-x86 architectures.")
+  | `x86 ->
+    warning "In 32-bit x86, arguments are passed through the stack.%!";
+    []
+  | #Arch.arm ->
+    let open ARM.CPU in
+    [r0; r1; r2; r3]
+  | a ->
+    let err = Format.sprintf "input_regs: Input registers have not been \
+                              implemented for %s." (Arch.to_string a) in
+    raise (Not_implemented err)
```

Comment:
I wonder if there's some more BAP idiomatic way of raising an error.

On another note, I wonder if we could just print a warning here and charge on regardless.